homeassistant/components/light/mqtt_template.py

```diff
@@ -152,6 +161,20 @@ def state_received(topic, payload, qos):
                 except ValueError:
                     _LOGGER.warning('Invalid color value received')
 
+            # read effect
+            if self._templates[CONF_EFFECT_TEMPLATE] is not None:
+                try:
+                    effect = self._templates[CONF_EFFECT_TEMPLATE].\
+                        render_with_possible_json_value(payload)
+                except ValueError:
```

Comment:
I'm not familiar with templates rendering, if there is no `ValueError` in templates rendering then yes this is useless.